src/tabpfn/model_loading.py

```diff
@@ -489,6 +490,17 @@ def load_model_criterion_config(
         )
     )
 
+    # Even though model_path can be a list of paths, we currently
+    # only use a single path
+    if len(resolved_model_paths) == 1:
+        # Only set model config if the model path is in the list of available model
+        # filenames to avoid passing arbitrary paths containing e.g. PII
+        _filenames = _get_model_filenames(which, model_version.value)
+
+        _path = resolved_model_paths[0]
+        if _path.name in _filenames:
+            set_model_config(_path.name, model_version.value)
```

Comment:
What about putting this in a function `_log_model_paths()` to reduce the complexity of this one?